src/openms/source/APPLICATIONS/TOPPBase.cpp

```diff
@@ -387,6 +408,130 @@ namespace OpenMS
+
+        // touch file to create it and set initial modification time stamp
+        QFile f;
+        f.setFileName(version_file_name.toQString());
+        f.open(QIODevice::WriteOnly);
+        f.close();
+        first_run = true;
+      }
+      
+      if (File::readable(version_file_name))
+      {
+        QDateTime last_modified_dt = QFileInfo(version_file_name.toQString()).lastModified();
+        QDateTime current_dt = QDateTime::currentDateTime();
+
+        // check if at least one day passed sincle last request
+        if (first_run || current_dt > last_modified_dt.addDays(1))
+        {
+          // update modification time stamp
+          struct stat old_stat;
+          time_t mtime;
+          struct utimbuf new_times;
+          stat(version_file_name.c_str(), &old_stat);
+          mtime = old_stat.st_mtime; 
+          new_times.actime = old_stat.st_atime; // keep accessuib time unchanged 
+          new_times.modtime = time(NULL);  // mod time to current time
+          utime(version_file_name.c_str(), &new_times);          
+
+          LOG_INFO << "The OpenMS team is collecting use statistics for quality control and funding purposes." << endl;
+          LOG_INFO << "We will never give out your personal data but you may disable this functionality by setting the environmental variable OPENMS_DISABLE_USAGE_STATISTICS." << endl;
+          
+          // Usage of a QCoreApplication is overkill here (and ugly too), but we just use the
+          // QEventLoop to process the signals and slots and grab the results afterwards from
+          // the MascotRemotQuery instance
+          char** argv2 = const_cast<char**>(argv);
+          QCoreApplication event_loop(argc, argv2);
+          NetworkGetRequest* query = new NetworkGetRequest(&event_loop);
+          query->setUrl(QUrl(QString("http://openms-update.informatik.uni-tuebingen.de/check/") + tool_version_string.toQString()));
+          QObject::connect(query, SIGNAL(done()), &event_loop, SLOT(quit()));
+          QTimer::singleShot(1000, query, SLOT(run()));          
+          QTimer::singleShot(5000, query, SLOT(timeOut()));
+          event_loop.exec();
+
+          if (!query->hasError())
+          {
+            writeLog_("Connecting to REST server successful. ");
+            QString response = query->getResponse();
+            VersionInfo::VersionDetails server_version = VersionInfo::VersionDetails::create(response);
+            if (server_version != VersionInfo::VersionDetails::EMPTY)
+            {
+              if (VersionInfo::getVersionStruct() < server_version)
+              {
+                writeLog_("Version " + version_ + " of " + tool_name_ + " is available at www.OpenMS.de");
+              }
+            }
+          }
+          else
+          {
+            writeLog_("Connecting to REST server failed. ");
+            writeLog_("Error: " + query->getErrorString());
+          }
+          delete query;
+        }
+      }
+    }
```

Comment:
/SYSTEM preferrably

or maybe CONCEPT ?!

Am 3/21/2016 um 2:55 PM schrieb Timo Sachsenberg:
> In src/openms/source/APPLICATIONS/TOPPBase.cpp
> <https://github.com/OpenMS/OpenMS/pull/1875#discussion_r56824872>:
>
>> +            {
>> +              if (VersionInfo::getVersionStruct() < server_version)
>> +              {
>> +                writeLog_("Version " + version_ + " of " + tool_name_ + " is available at www.OpenMS.de");
>> +              }
>> +            }
>> +          }
>> +          else
>> +          {
>> +            writeLog_("Connecting to REST server failed. ");
>> +            writeLog_("Error: " + query->getErrorString());
>> +          }
>> +          delete query;
>> +        }
>> +      }
>> +    }
>
> where would you put it preferably?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenMS/OpenMS/pull/1875/files/5e427b0181888381fa8d39484278e8d1af1ee40c#r56824872>
>